src/TreeHouse/EventStore/DBALEventStore.php

```diff
@@ -52,22 +52,40 @@ public function __construct(Connection $connection, SerializerInterface $seriali
      */
     public function getStream($id)
     {
-        $query = 'SELECT uuid, name, payload, payload_version, version, datetime_created
-                  FROM event_store
-                  WHERE uuid = :uuid
-                  ORDER BY version ASC';
+        $stream = $this->getPartialStream($id, 0);
 
-        $statement = $this->connection->prepare($query);
-
-        $statement->bindValue('uuid', $id);
-        $statement->execute();
+        if (!$stream->count()) {
+            throw new EventStreamNotFoundException($id);
+        }
 
-        $eventsFromStore = $statement->fetchAll();
+        return $stream;
+    }
 
-        if (empty($eventsFromStore)) {
-            throw new EventStreamNotFoundException($id);
+    /**
+     * @inheritdoc
+     */
+    public function getPartialStream($id, $fromVersion, $toVersion = null)
+    {
+        $qb = $this->connection->createQueryBuilder()
+            ->select('uuid, name, payload, payload_version, version, datetime_created')
+            ->from('event_store')
+            ->where('uuid = :uuid')
+            ->andWhere('version > :version_from')
+            ->orderBy('version', 'ASC')
+
+            ->setParameter('uuid', $id)
+            ->setParameter('version_from', $fromVersion)
+        ;
+
+        if ($toVersion) {
+            $qb
+                ->andWhere('version <= :version_to')
+                ->setParameter('version_to', $toVersion)
+            ;
         }
 
+        $eventsFromStore = $qb->execute()->fetchAll();
```

Comment:
Not in the scope of this PR but this seems to be pretty inefficient with a large collection of events.